bin/build.sh

```diff
@@ -1,20 +1,39 @@
 #!/bin/bash
 set -ex
 
+# branch names don't correspond 1:1 with config file names
+branch="$1"
+case branch in
+    --dev)
```

Comment:
you're looking for `--dev` on the CLI, but in the gitlab YAML you're just passing the branch name. I think here you only want e.g. `dev)` and `master)` without the leading `--`.